File: cracking coding interviews/anagram_min_char.js
function main() {
    let a = readLine();
    let b = readLine();
    let obj = {},
        count = 0;
    for (let i = 0; i < a.length; i++) {
        const o = a[i];
        obj[o] = obj[o] ? obj[o] + 1 : 1;
    }

    for (let i = 0; i < b.length; i++) {
        const o = b[i];
        obj[o] = obj[o] ? obj[o] - 1 : -1;
    }

    Object.entries(obj).forEach(([key, value]) => {
        count += Math.abs(value);
    });
    console.log(count);
}

File: cracking coding interviews/anagram.min_char.py
def number_needed(a, b):
    count = 0; mp = {}
    for i in a:
        if i in mp:
            mp[i] += 1
        else:
            mp[i] = 1
    
    for j in b:
        if j in mp:
            mp[j] -= 1
        else:
            mp[j] = -1
        
    for k, v in mp.items():
        count = count + abs(v)
        
    return count
        

a = input().strip()
b = input().strip()

print(number_needed(a, b))
